### PR TITLE
Implemented easyComputerFire and Refactored placeShip

### PR DESCRIPTION
**Pull request by srb76**
-Added three private vars for easyComputerFire() method
-Added easyComputerFire(), a method that makes the ai fire at every
other space
-Added a test for easyComputerFire
--(FIXED) This has one minor bug in that the computer will fire at sunk clipper locations. This could cause the game to end prematurely in defeat for the player. 

Edit 3/16:
Refactored placeShip
-Reorgranized code to be more readable, moved assignments and variable
initialization to top
-Restructured logic so that the same code is not repeated for both
horizontal and vertical
-Removed checking for civilian or military ship based on ship's name.
Changed this to check the type of the ship gotten from the ship's name.
--This method could be divided further if anyone feels it is necessary

Fixed bug with easyComputerFire
-Now checks for duplicate firing locations and will attempt another fire
until it is not hitting a duplicate
-Updated test to have full coverage and account for changes

### CODE REVIEW

**Comment by tnoelcke** [comment]
Whom ever was supposed to do this code review will have to do it after the pull request is Merged because i need this code tonight so i can finish this project up.

File: src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java [hunk 1]
```diff
@@ -269,85 +274,90 @@ public boolean shipInBounds(Coordinate c, int orientation, int length){
...
-                playerMilitaryShips[militaryPlaceIndex] = (MilitaryShip) getShip(shipName);
-                militaryPlaceIndex++;
-            }
-            else{
-                getShip(shipName).setLocation(start, end);
-                getShip(shipName).setPoints(myPoints);
-                playerCivilianShips[civilianPlaceIndex] = (CivilianShip) getShip(shipName);
-                civilianPlaceIndex++;
-            }
         }
```
Comment (GarrettHaley) [inline src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java]: I appreciate comments you made here. It's really clear what the code does, and the changes you made.

Comment (GarrettHaley) [inline src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java]: You've refactored a lot here. This section was especially bad, and the changes you have made a definitely paramount to cleaning up our code.

File: src/Test/BattleshipModelTest.java [hunk 2]
```diff
@@ -212,6 +212,35 @@ public void smartShootAtPlayer(){
 
     }
 
+    @Test
+    public void easyComputerFire(){
```
Comment (GarrettHaley) [inline src/Test/BattleshipModelTest.java:216]: This test is really clean. The code you've produced here is probably the best I've seen so far on our project. I appreciate it. (in easyComputerFire())

File: src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java [hunk 3]
```diff
@@ -269,85 +274,90 @@ public boolean shipInBounds(Coordinate c, int orientation, int length){
...
+        placingShip.setLocation(start,end);
+        placingShip.setPoints(myPoints);
+
+        //Military
+        if (placingShip instanceof MilitaryShip) {
+            playerMilitaryShips[militaryPlaceIndex] = (MilitaryShip)placingShip;
+            militaryPlaceIndex++;
+        }
+        //Civilian
+        else if (placingShip instanceof CivilianShip) {
```
Comment (GarrettHaley) [inline src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java:353]: This probably could just be an else statement. Its fine either way, but that might be a bit cleaner if you change it.

File: src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java [hunk 4]
```diff
@@ -269,85 +274,90 @@ public boolean shipInBounds(Coordinate c, int orientation, int length){
...
+            playerCivilianShips[civilianPlaceIndex] = (CivilianShip)placingShip;
+            civilianPlaceIndex++;
+        }
+
+        //Ship was placed successfully, return null
         return null;
+
     }
 
     public String shootAtComputer(int row, int col) {
```
Comment (GarrettHaley) [inline src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java:363]: I'm a little confused to why the function on line 363 is still there. I think we may have forgotten to delete it. If we can, we should probably delete this function that isn't used before we turn in the project.

File: src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java [hunk 5]
```diff
@@ -538,6 +548,49 @@ public void shootAtPlayer(int across, int down) {
 
     }
 
+    public void easyComputerFire() {
```
Comment (GarrettHaley) [inline src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java:551]: The code in this function is very clean. I would say that you may want to be more consistent on your spacing, and reduce the amount a little. The code is a little too spread out.

File: src/main/java/edu/oregonstate/cs361/battleship/Main.java [hunk 6]
```diff
@@ -76,7 +76,8 @@ private static String fireAt(Request req, Response res) {
             res.status(400);
             return result;
         } else //(result == null)
-            currModel.smartShootAtPlayer();
+            //currModel.smartShootAtPlayer();
```
Comment (GarrettHaley) [inline src/main/java/edu/oregonstate/cs361/battleship/Main.java:79]: This comment should be deleted or changed. It's actually a little confusing, as you've deleted currModel.smartShootAtPlayer(). If you wanted to comment on what that line of code used to be, maybe add that in. For example, "This used to be currModel.smartShootAtPlayer()" .

**Review by GarrettHaley — COMMENTED** [review]
Overall, really great work. There are some minor changes that should be looked at. The biggest being the function which is not used at all in our program.

File: src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java [hunk 7]
```diff
@@ -269,85 +274,90 @@ public boolean shipInBounds(Coordinate c, int orientation, int length){
...
-            else{
-                getShip(shipName).setLocation(start, end);
-                getShip(shipName).setPoints(myPoints);
-                playerCivilianShips[civilianPlaceIndex] = (CivilianShip) getShip(shipName);
-                civilianPlaceIndex++;
-            }
         }
 
+        //Return error string if ship would go off board
+        if(endAcross > BOARD_SIZE || endDown > BOARD_SIZE)
```
Comment (GarrettHaley) [inline src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java:316]: I appreciate comments you made in this function (example above). It's really clear what the code does, and the changes you made.

File: src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java [hunk 8]
```diff
@@ -269,85 +274,90 @@ public boolean shipInBounds(Coordinate c, int orientation, int length){
     }
 
 
-
+/*Checks if ship placement at specified location would be inbounds, not overlap any other ships, and then
+updates the cooresponding arrays with the ships points */
     public String placeShip(String shipName, String AcrossS, String DownS, String orientation) {
```
Comment (GarrettHaley) [inline src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java:279]: You've refactored a lot here. This section was especially bad, and the changes you have made a definitely paramount to cleaning up our code.

**Review by GarrettHaley — COMMENTED** [review]
Adjusted two comments so they could be more easily seen.

**Pull request merged**